src/backend/distributed/master/master_citus_tools.c

```diff
@@ -229,28 +229,37 @@ ExecuteCommandsInParallelAndStoreResults(StringInfo *nodeNameArray, int *nodePor
 		palloc0(commmandCount * sizeof(MultiConnection *));
 	int finishedCount = 0;
 
-	/* establish connections */
+	/* start connections asynchronously */
 	for (commandIndex = 0; commandIndex < commmandCount; commandIndex++)
 	{
 		char *nodeName = nodeNameArray[commandIndex]->data;
 		int nodePort = nodePortArray[commandIndex];
 		int connectionFlags = FORCE_NEW_CONNECTION;
```

Comment:
while you are at it. Could you also check if  we can safely re-use existing connections, or let the connection stay open afterwards.

this could enable setting session level gucs at workers

Reply:
I talked with @onderkalaci about this. Making this session-wide will encourage people to use it more, and `run_command_on_shards` doesn't work well with transactional semantics, deadlock detection, etc.

Also since we are unaware what is sent as the command, this will break `FOR_DML`, `FOR_DDL`, etc.